src/quantities.hxx

```diff
@@ -177,14 +166,220 @@ auto m_vis(auto &df, const std::string &outputname,
+ \hat{\zeta} \f] where \f$\vec{p}_\text{T}^{p_{1,2}}\f$ corresponds to the
+ transverse momentum vector of the first (second) lepton and \f$\hat{\zeta}\f$
+ to the bisectional direction between the two leptons in the transverse plane.
+ For more information check
+
+ D. Jang, “Search for MSSM Higgs decaying to
+ tau pairs in pp collision at √s=1.96 TeV at CDF”. PhD thesis, Rutgers
+ University, 2006. FERMILAB-THESIS-2006-11.
+
+ * @param df the input dataframe
+ * @param p_1_p4 the lorentz vector of the first particle
+ * @param p_2_p4 the lorentz vector of the second particle
+ * @param met the lorentz vector of the met
+ * @param outputname the name of the new column containing the pZetaMissVis
+ value
+ * @return a new dataframe with the new column
+ */
+auto pzetamissvis(auto &df, const std::string &outputname,
+                  const std::string &p_1_p4, const std::string &p_2_p4,
+                  const std::string &met) {
+    float alpha = 0.85;
+    auto calculate_pzetamissvis = [alpha](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                                          ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                                          ROOT::Math::PtEtaPhiMVector &met) {
+        auto met_3dvec = met.Vect();
+        met_3dvec.SetZ(0.0);
+        // calculate zeta for the delepton system
+        auto p1_norm = p_1_p4.Vect().Unit();
+        auto p2_norm = p_2_p4.Vect().Unit();
+        p1_norm.SetZ(0.0);
+        p2_norm.SetZ(0.0);
+        p1_norm = p1_norm.Unit();
+        p2_norm = p2_norm.Unit();
+        auto zeta = (p1_norm + p2_norm).Unit();
+
+        auto dileptonsystem = p_1_p4.Vect() + p_2_p4.Vect();
+        dileptonsystem.SetZ(0);
+        auto pzetaVis = dileptonsystem.Dot(zeta);
+        return met_3dvec.Dot(zeta) - (alpha * pzetaVis);
+    };
+    return df.Define(outputname, calculate_pzetamissvis, {p_1_p4, p_2_p4, met});
+}
+/**
+ * @brief function used to calculate mTdileptonMET, which is the transverse mass
+ * of the di-lepton system. The transverse mass is calculated using the
+ * vectoroperations::calculateMT function.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the mTdileptonMET value
+ * @param p_1_p4 lorentz vector of the first particle
+ * @param p_2_p4 lorentz vector of the second particle
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+auto mTdileptonMET(auto &df, const std::string &outputname,
+                   const std::string &p_1_p4, const std::string &p_2_p4,
+                   const std::string &met) {
+    auto calculate_mTdileptonMET = [](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                                      ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                                      ROOT::Math::PtEtaPhiMVector &met) {
+        auto dileptonsystem = p_1_p4 + p_2_p4;
+        return vectoroperations::calculateMT(dileptonsystem, met);
```

Comment:
better write the sum directly into the function call?

Reply:
changed